internal/observability/metrics/tools/benchmark/benchmark.go

```diff
@@ -0,0 +1,285 @@
+	appliedScenarioCount, err := m.Int64ObservableCounter(
+		appliedScenarioCount,
+		metrics.WithDescription(appliedScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+	runningScenarioCount, err := m.Int64ObservableCounter(
+		runningScenarioCount,
+		metrics.WithDescription(runningScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+	completeScenarioCount, err := m.Int64ObservableCounter(
+		completeScenarioCount,
+		metrics.WithDescription(completeScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+
+	appliedBenchJobCount, err := m.Int64ObservableCounter(
+		appliedBenchmarkJobCount,
+		metrics.WithDescription(appliedScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+	runningBenchJobCount, err := m.Int64ObservableCounter(
+		runningBenchmarkJobCount,
+		metrics.WithDescription(runningScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+	completeBenchJobCount, err := m.Int64ObservableCounter(
+		completeBenchmarkJobCount,
+		metrics.WithDescription(completeScenarioCountDescription),
+		metrics.WithUnit(metrics.Dimensionless),
+	)
+	if err != nil {
+		return err
+	}
+
+	// appliedJobCount, err := m.Int64ObservableCounter(
+	// 	appliedJobCount,
+	// 	metrics.WithDescription(appliedJobCountDescription),
+	// 	metrics.WithUnit(metrics.Dimensionless),
+	// )
+	// if err != nil {
+	// 	return err
+	// }
+	// runningJobCount, err := m.Int64ObservableCounter(
+	// 	runningJobCount,
+	// 	metrics.WithDescription(runningJobCountDescription),
+	// 	metrics.WithUnit(metrics.Dimensionless),
+	// )
+	// if err != nil {
+	// 	return err
+	// }
+	// completeJobCount, err := m.Int64ObservableCounter(
+	// 	completeBenchmarkJobCount,
+	// 	metrics.WithDescription(completeScenarioCountDescription),
+	// 	metrics.WithUnit(metrics.Dimensionless),
+	// )
+	// if err != nil {
+	// 	return err
+	// }
+
+	_, err = m.RegisterCallback(
+		func(_ context.Context, o api.Observer) error {
+			// scenario status
+			sst := map[string]int64{
+				applied:  0,
+				running:  0,
+				complete: 0,
+			}
+			for k, v := range om.op.GetScenarioStatus() {
+				sst[applied] += v
+				if k == v1.BenchmarkScenarioCompleted {
+					sst[complete] += v
+				} else {
+					sst[running] += v
+				}
+			}
+			o.ObserveInt64(appliedScenarioCount, sst[applied])
+			o.ObserveInt64(runningScenarioCount, sst[running])
+			o.ObserveInt64(completeScenarioCount, sst[complete])
+
+			// benchmark job status
+			bst := map[string]int64{
+				applied:  0,
+				running:  0,
+				complete: 0,
+			}
+			for k, v := range om.op.GetBenchmarkJobStatus() {
+				bst[applied] += v
+				if k == v1.BenchmarkJobCompleted {
+					bst[complete] += v
+				} else {
+					bst[running] += v
```

Comment:
If `k` is `v1.BenchmarkJobNotReady`, etc., is it also counted as `running`? 🤔

Reply:
It is a temporary and will be fixed by the next task for bugfix